pkg/kubelet/dockertools/manager.go

```diff
@@ -780,7 +780,16 @@ type dockerVersion struct {
 	*semver.Version
 }
 
+// newDockerVersion returns a semantically versioned docker version value
 func newDockerVersion(version string) (dockerVersion, error) {
+	// Older versions of Docker could return non-semantically versioned values (distros like Fedora
+	// included partial values such as 1.8.1.fc21 which is not semver). Force those values to be semantic.
+	if strings.Count(version, ".") > 2 {
+		switch parts := strings.Split(version, "."); {
+		case len(parts) >= 3:
+			version = strings.Join(parts[:3], ".") + "-" + strings.Join(parts[3:], ".")
```

Comment:
May be I am misunderstanding something, but doesn't it mutate a correct string x.y.z-prerelease.n to something x.y.z-prerelease-n? 

For example, I tested it with {value: "1.8.1-beta.12", out: "1.8.1-beta.12"} and it failed because the out form was  1.8.1-beta-12.